parsers.js

```diff
@@ -0,0 +1,143 @@
+const _ = require("lodash");
+
+function object(value) {
+  if (_.isObject(value)) { return value; }
+  if (_.isString(value)) {
+    try {
+      return JSON.parse(value);
+    } catch (e) {
+      throw new Error(`Couldn't parse provided value as object: ${value}`);
+    }
+  }
+  throw new Error(`${value} is not a valid object`);
+}
+
+function number(value) {
+  const validNumber = (value) => _.isNumber(value) && _.isFinite(value) && !_.isNaN(value);
+  if (validNumber(value)) {
+    return value;
+  }
+  if (validNumber(parseFloat(value))) {
+    return parseFloat(value);
+  }
+  throw new Error(`Value ${value} is not a valid number`);
+}
+
+function boolean(value) {
+  if (_.isNil(value)) { return false; }
+  if (_.isBoolean(value)) { return value; }
+  if (_.isString(value) && _.isEmpty(value)) { return false; }
+  if (
+    _.isString(value)
+    && _.includes(["true", "false"], value.toLowerCase().trim())
+  ) {
+    return value.toLowerCase().trim() === "true";
+  }
```

Comment:
```suggestion
  if (_.isString(value)) {
    const stringValue = value.toLowerCase().trim();
    if (["", "false"].includes(stringValue)) { return false; }
    if (stringValue === "true" ) { return true }
  }
```

Afaik `isEmpty` does not process strings: https://lodash.com/docs/4.17.15#isEmpty
Also `_.includes` is a multipurpose function while we only need to compare against a known array, thus it can be achieved with a much simpler call.

Reply:
isEmpty works fine for strings 
![image](https://user-images.githubusercontent.com/92038478/158389553-9c124eb2-134a-4579-b9ce-71bacf4bf921.png)

I wanted to logically separate the cases in the code so that it's easily noticeable what's covered (nil, boolean, empty string, "true" / "false" string), but I have no problem with merging two "string" cases together.